Neos.Flow/Tests/Unit/Http/Middleware/MethodOverrideMiddlewareTest.php

```diff
@@ -0,0 +1,107 @@
+
+    /**
+     * @var ResponseInterface|MockObject
+     */
+    private $mockResponse;
+
+    public function setUp(): void
+    {
+        $this->middleware = new MethodOverrideMiddleware();
+
+        $this->mockRequest = $this->getMockBuilder(ServerRequestInterface::class)->getMock();
+        $this->mockRequestHandler = $this->getMockBuilder(RequestHandlerInterface::class)->getMock();
+        $this->mockResponse = $this->getMockBuilder(ResponseInterface::class)->getMock();
+    }
+
+    public function process_matchingRequests_dataProvider(): \Traversable
+    {
+        yield 'parsedBody (__method)' => ['method' => 'POST', 'headers' => [], 'parsedBody' => ['__method' => 'PUT'], 'expectedMethod' => 'PUT'];
+        yield 'header (X-Http-Method-Override)' => ['method' => 'POST', 'headers' => ['X-Http-Method-Override' => 'PATCH'], 'parsedBody' => [], 'expectedMethod' => 'PATCH'];
+        yield 'header (X-Http-Method header)' => ['method' => 'POST', 'headers' => ['X-Http-Method' => 'DELETE'], 'parsedBody' => [], 'expectedMethod' => 'DELETE'];
+        yield 'parsedBody and X-Http-Method header' => ['method' => 'POST', 'headers' => ['X-Http-Method' => 'DELETE'], 'parsedBody' => ['__method' => 'PUT'], 'expectedMethod' => 'PUT'];
+        yield 'X-Http-Method-Override and X-Http-Method header' => ['method' => 'POST', 'headers' => ['X-Http-Method-Override' => 'PATCH', 'X-Http-Method' => 'DELETE'], 'parsedBody' => [], 'expectedMethod' => 'PATCH'];
+        yield 'parsedBody and both headers' => ['method' => 'POST', 'headers' => ['X-Http-Method-Override' => 'PATCH', 'X-Http-Method' => 'DELETE'], 'parsedBody' => ['__method' => 'PUT'], 'expectedMethod' => 'PUT'];
+    }
+
+    /**
+     * @test
+     * @dataProvider process_matchingRequests_dataProvider
+     */
+    public function process_matchingRequests(string $method, array $headers, $parsedBody, string $expectedMethod): void
+    {
+        $this->mockRequest->method('getMethod')->willReturn($method);
+        $this->mockRequest->method('getParsedBody')->willReturn($parsedBody);
+        $this->mockRequest->method('hasHeader')->willReturnCallback(fn($header) => isset($headers[$header]));
+        $this->mockRequest->method('getHeaderLine')->willReturnCallback(fn($header) => $headers[$header]);
+
+        $mockAlteredRequest = $this->getMockBuilder(ServerRequestInterface::class)->disableOriginalConstructor()->getMock();
+        $this->mockRequest->expects(self::once())->method('withMethod')->with($expectedMethod)->willReturn($mockAlteredRequest);
+        $this->mockRequestHandler->expects(self::once())->method('handle')->willReturnCallback(function($request) use ($mockAlteredRequest) {
+            self::assertSame($request, $mockAlteredRequest);
+            return $this->mockResponse;
+        });
+
+        $this->middleware->process($this->mockRequest, $this->mockRequestHandler);
+    }
+
+    public function process_nonMatchingRequests_dataProvider2(): \Traversable
```

Comment:
```suggestion
    public function nonMatchingRequests_dataProvider2(): \Traversable
```
Doesn't it make more sense to name this after *what* it provides, rather than *for whom*? I'd even be inclined to strip off the `_dataProvider` alltogether.